API-Samples/Java/app/src/main/java/apisamples/TokensStore.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    public String getOrCreate(String key, Supplier<TimedEntry> valueSupplier) {
+        TimedEntry timedEntry = cache.getIfPresent(key);
+        if (timedEntry == null || timedEntry.isExpired()) {
+            timedEntry = valueSupplier.get();
+            cache.put(key, timedEntry);
+        }
+        return timedEntry.value;
+    }
+
+    /**
+     * Generate new Arm Access Token
+     * @return the Arm Access Token
+     */
+    private TimedEntry generateArmAccessToken() {
+
+        var tokenRequestContext = new TokenRequestContext();
+        tokenRequestContext.addScopes(String.format("%s/.default", AzureResourceManager));
+
+        DefaultAzureCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+        AccessToken accessToken = Objects.requireNonNull(defaultCredential.getToken(tokenRequestContext).block());
+        //Calculate Expiration token as Expiry value - Buffer Time ( default 5 Min)
+        long expiryEpocTime = accessToken.getExpiresAt().toInstant().toEpochMilli() - ExpiryBufferTimeInMs;
+        return new TimedEntry(accessToken.getToken(),expiryEpocTime);
+    }
+
+    /**
+     * Generate Video Indexer Access Token according to Permission and Scope
+     * we assume Project ID and Video Id are not used in this sample
+     * @return the Video Indexer Account  Access Token according to this Client Permission + Scope ( Default : Contributor /Account)
+     */
+    private TimedEntry generateVIAccessToken() {
+
+        var accessTokenRequest = new AccessTokenRequest(permission, scope);
+        var accessTokenRequestStr = gson.toJson(accessTokenRequest);
+
+        var requestUri = MessageFormat.format("{0}/subscriptions/{1}/resourcegroups/{2}/providers/Microsoft.VideoIndexer/accounts/{3}/generateAccessToken?api-version={4}",
+                AzureResourceManager, SubscriptionId, ResourceGroup, AccountName, ApiVersion);
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI(requestUri))
+                    .headers("Content-Type", "application/json;charset=UTF-8")
+                    .headers("Authorization", "Bearer " + this.getArmAccessToken())
+                    .POST(HttpRequest.BodyPublishers.ofString(accessTokenRequestStr))
+                    .build();
+
+            var response = httpStringResponse(request);
+            AccessTokenResponse accessTokenResponse = gson.fromJson(response.body(), AccessTokenResponse.class);
+            String token = accessTokenResponse.accessToken;
+            DecodedJWT jwt = JWT.decode(token);
+            //Calculate Expiration token as Expiry value - Buffer Time ( default 5 Min)
+            long expiryEpocTime = jwt.getExpiresAtAsInstant().toEpochMilli() - ExpiryBufferTimeInMs;
+            return new TimedEntry(accessTokenResponse.accessToken, expiryEpocTime);
+        } catch (URISyntaxException | IOException | InterruptedException ex) {

```

Comment:
What is the point of the catch? I think you can remove it

Reply:
if you remove it the code will not compile